kafka/producer.go

```diff
@@ -27,11 +27,30 @@ import (
 #include <stdlib.h>
 #include <librdkafka/rdkafka.h>
 
-rd_kafka_resp_err_t do_produce (rd_kafka_topic_t *rkt, int32_t partition,
-          int msg_flags,
-          void *val, size_t val_len, void *key, size_t key_len, uintptr_t cgoid) {
-  return rd_kafka_produce(rkt, partition, msg_flags, val, val_len, key, key_len,
-                          (void *)cgoid);
+rd_kafka_resp_err_t do_produce (rd_kafka_t *rk,
+          rd_kafka_topic_t *rkt, int32_t partition,
+          int msgflags,
+          void *val, size_t val_len, void *key, size_t key_len,
+          int64_t timestamp,
+          uintptr_t cgoid) {
+
+#if RD_KAFKA_VERSION >= 0x00090300
+  return rd_kafka_producev(rk,
+        RD_KAFKA_V_RKT(rkt),
```

Comment:
Any reason for using the rkt version here but in Python you passed the topic string?